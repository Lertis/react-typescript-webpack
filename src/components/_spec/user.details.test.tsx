import React from 'react';
import renderer from "react-test-renderer"
import { shallow } from 'enzyme';
import PersonDetail from '../person.info';
import { cloneDeep } from "lodash";
import { USER_INFO_1, USER_INFO_2 } from "./user.details.set";
import "./../../setupTests";

describe("PersonDetail Component", () => {
	it(`should be created with: with userDetails and nothingToShow as false`, () => {
		let component = renderer.create(
			<PersonDetail detail={cloneDeep(USER_INFO_1)} nothingToShow={false}></PersonDetail>,
		);
		let detailsComponent = component.toJSON();
		expect(detailsComponent).toMatchSnapshot();
	});

	it('[Method: anyUserDetails]', () => {
		const wrapper = shallow(<PersonDetail detail={cloneDeep(USER_INFO_1)} nothingToShow={false}></PersonDetail>);
		const instance = wrapper.instance();
		const res = (instance as any).anyUserDetails();
		expect(res).toMatchSnapshot();
	});

	it('componentWillReceiveProps', () => {
		const component = shallow(<PersonDetail detail={cloneDeep(USER_INFO_1)} nothingToShow={false}></PersonDetail>);
		component.setProps({
			detail: cloneDeep(USER_INFO_2)
		});
		expect(component).toMatchSnapshot();
		expect(component).toBeTruthy();
	})
});
