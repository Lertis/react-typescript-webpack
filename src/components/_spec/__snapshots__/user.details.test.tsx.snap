// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonDetail Component [Method: anyUserDetails] 1`] = `
<WithStyles(ForwardRef(Card))>
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Typography))
      color="textSecondary"
      gutterBottom={true}
    >
      User e-mail : 
      
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      component="h2"
      variant="h5"
    >
      User name: 
      
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      color="textSecondary"
    >
      Company: 
      
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      className="flex-stretch"
      component="p"
      variant="body2"
    >
      <Memo(PhoneIcon) />
       
      
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(CardContent))>
</WithStyles(ForwardRef(Card))>
`;

exports[`PersonDetail Component componentWillReceiveProps 1`] = `ShallowWrapper {}`;

exports[`PersonDetail Component should be created with: with userDetails and nothingToShow as false 1`] = `
<div>
  <div>
    No users to be displayed!
  </div>
</div>
`;
